src/ReactPlaceholder.js

```diff
@@ -80,14 +83,28 @@ export default class ReactPlaceholder extends React.Component {
   }
 
   render() {
-    return this.state.ready ? this.props.children : this.getFiller();
+    const { ready, holdPlaceholder } = this.state;
+    return ready && !holdPlaceholder ? this.props.children : this.getFiller();
   }
 
   componentWillReceiveProps(nextProps) {
-    if (!this.props.firstLaunchOnly && this.state.ready && !nextProps.ready) {
-      this.setNotReady();
+    if (!this.props.firstLaunchOnly && !nextProps.ready) {
```

Comment:
I find this new code over complicated and difficult to read...
Why didn't you use the two functions `setReady` and `setNotReady` already implemented?
we already have a state variable called `ready` to decide if we should render the placeholder or the content => I don't see the need of the new `this.state.holdPlaceholder` variable